manager/dispatcher/dispatcher.go

```diff
@@ -572,20 +573,45 @@ func (d *Dispatcher) Tasks(r *api.TasksRequest, stream api.Dispatcher_TasksServe
 			return err
 		}
 
-		select {
-		case event := <-nodeTasks:
-			switch v := event.(type) {
-			case state.EventCreateTask:
-				tasksMap[v.Task.ID] = v.Task
-			case state.EventUpdateTask:
-				tasksMap[v.Task.ID] = v.Task
-			case state.EventDeleteTask:
-				delete(tasksMap, v.Task.ID)
+		// bursty events should be processed in batches and sent out snapshot
+		const eventBatchLimit = 200
+		const eventPausedGap = 50 * time.Millisecond
+		var eventCnt int
+		// eventPaused is true when there are events
+		// but next event has not arrived within eventPausedGap
+		eventPaused := false
+
+		for eventCnt < eventBatchLimit && !eventPaused {
+			select {
+			case event := <-nodeTasks:
+				switch v := event.(type) {
+				case state.EventCreateTask:
+					tasksMap[v.Task.ID] = v.Task
+					eventCnt++
+				case state.EventUpdateTask:
+					if oldTask, exists := tasksMap[v.Task.ID]; exists {
+						if apiutils.TasksSpecEqual(oldTask, v.Task) {
+							// task spec has not changed
+							// this update would not trigger action at agent
+							tasksMap[v.Task.ID] = v.Task
```

Comment:
I don't see how this branch of code does anything different, except not increment the `eventCnt`.

The right approach here would be to keep a copy of the last assignment set sent. If the assignment set is not different, then we don't send it.